web/src/layout/leftpanel/mobileinfotab.js

```diff
@@ -20,6 +20,27 @@ const MobileInfoTab = () => {
     return <Redirect to={{ pathname: '/map', search: location.search }} />;
   }
 
+  const [mobileAppVersion, setMobileAppVersion] = useState(null);
+  // Check app version once
+  if (!mobileAppVersion && window.isCordova) {
+    codePush.getCurrentPackage((localPackage) => {
+      if (!localPackage) {
+        console.log('CodePush: No updates have been installed yet');
+        setMobileAppVersion(null);
+        return;
+      }
+
+      const {
+        appVersion, // The native version of the application this package update is intended for.
+        description, // same as given during deployment
+        // isFirstRun, // flag indicating if the current application run is the first one after the package was applied.
+        label, // The internal label automatically given to the update by the CodePush server, such as v5. This value uniquely identifies the update within it's deployment
+      } = localPackage;
+
+      setMobileAppVersion(`${appVersion} ${label} (${description})`);
+    }, err => console.error(err));
+  }
+
```

Comment:
Would this be slightly better with a `useEffect`? 

i.e. instead of
```js
if (!mobileAppVersion && window.isCordova) { 
  ...
}
```
It can be
```js
useEffect(() => {
  if (window.isCordova) { 
    ...
  }
}, []);
```

Reply:
I'm not that familiar with `useEffect` - what difference would it have?
Happy to do the change either way ;-)

Reply:
I don't think that it makes a big difference here, but the current if-statement gets evaluated on every render, which means that if `codePush.getCurrentPackage` is slow, then it could potentially get triggered multiple times. `useEffect` with an empty array will make sure that it only runs on the first render.

My brain also reads the `useEffect` with an empty array like the `onComponentDidMount`, so it's a bit easier for me to understand when it's supposed to run. But that's just a personal preference :-)

Reply:
understood! Did the changes.